easybuild/easyconfigs/a/AlphaFold/AlphaFold-2.3.1-foss-2022a-CUDA-11.7.0.eb

```diff
@@ -0,0 +1,169 @@
+easyblock = 'PythonBundle'
+
+name = 'AlphaFold'
+version = '2.3.1'
+versionsuffix = '-CUDA-%(cudaver)s'
+
+homepage = 'https://deepmind.com/research/case-studies/alphafold'
+description = "AlphaFold can predict protein structures with atomic accuracy even where no similar structure is known"
+
+toolchain = {'name': 'foss', 'version': '2022a'}
+
+builddependencies = [
+    # required for installing dm-tree
+    ('Bazel', '5.1.1'),
+    ('CMake', '3.23.1'),
+]
+
+dependencies = [
+    ('Python', '3.10.4'),
+    ('CUDA', '11.7.0', '', SYSTEM),
+    ('SciPy-bundle', '2022.05'),
+    ('PyYAML', '6.0'),
+    ('TensorFlow', '2.11.0', versionsuffix),
+    ('Biopython', '1.79'),
+    ('HH-suite', '3.3.0'),
+    ('HMMER', '3.3.2'),
+    ('Kalign', '3.3.5'),
+    ('jax', '0.3.14', versionsuffix),  # also provides absl-py
+    ('UCX-CUDA', '1.12.1', versionsuffix),
+    ('cuDNN', '8.4.1.50', versionsuffix, SYSTEM),
+    ('NCCL', '2.12.12', versionsuffix),
+    ('OpenMM', '8.0.0'),
```

Comment:
@maxim-masterov Doesn't `OpenMM` need to be GPU-capable for AlphaFold?

Reply:
@boegel I've added CUDA to OpenMM, It builds fine and passes all tests after specifying the OPENMM_CUDA_COMPILER variable. Without this variable, OpenMM tries to use `/usr/local/cuda/bin/nvcc`, instead of `${EBROOTCUDA}/bin/nvcc`. The only test that fails with this variable is `CudaCompiler`, which requires `OPENMM_CUDA_COMPILER` not to be set, so I've excluded it.

Reply:
I gave this a try, as we also had some requests for this new AlphaFold version, but I do get an internal compiler error as well:

```
/dev/shm/f115372/OpenMM/8.0.0/foss-2022a/openmm-8.0.0/platforms/common/src/CommonKernels.cpp: In member function void OpenMM::CommonCalcGayBerneForceKernel::sortAtoms():
/dev/shm/f115372/OpenMM/8.0.0/foss-2022a/openmm-8.0.0/platforms/common/src/CommonKernels.cpp:5055:6: internal compiler error: in vect_get_vec_defs_for_operand, at tree-vect-stmts.c:1450
 5055 | void CommonCalcGayBerneForceKernel::sortAtoms() {
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
0x69b359 vect_get_vec_defs_for_operand(vec_info*, _stmt_vec_info*, unsigned int, tree_node*, vec<tree_node*, va_heap, vl_ptr>*, tree_node*)
        ../../gcc/tree-vect-stmts.c:1450
0xf42df4 vect_build_gather_load_calls
        ../../gcc/tree-vect-stmts.c:2728
0xf42df4 vectorizable_load
        ../../gcc/tree-vect-stmts.c:8718
0xf4bca0 vect_transform_stmt(vec_info*, _stmt_vec_info*, gimple_stmt_iterator*, _slp_tree*, _slp_instance*)
        ../../gcc/tree-vect-stmts.c:10922
0xf4fa41 vect_transform_loop_stmt
        ../../gcc/tree-vect-loop.c:9254
0xf6740d vect_transform_loop(_loop_vec_info*, gimple*)
        ../../gcc/tree-vect-loop.c:9690
0xf9059c try_vectorize_loop_1
        ../../gcc/tree-vectorizer.c:1104
0xf91181 vectorize_loops()
        ../../gcc/tree-vectorizer.c:1243
Please submit a full bug report,
with preprocessed source if appropriate.
Please include the complete backtrace with any bug report.
See <https://gcc.gnu.org/bugs/> for instructions.
```

Reply:
odd, what `optarch` flags do you use? and what machine are you building on?

Reply:
I didn't set any `optarch`, so it was using `-march=native`, and this was on an AMD EPYC 7763. Also tried OpenMM 7.7.0 with CUDA and foss 2022a, but that resulted in the same error.

Reply:
I built succesfully on Intel Platinum 8360Y with `optarch=Intel:O2 -march=core-avx2;GCC:O2 -mavx2 -mfma` and NVIDIA A100. And commenting out `CUDA` allowed me to build on AMD EPYC 7H12 without GPUs on board.

Reply:
I confirm that just removing `-ftree-vectorize` is enough to workaround this ICE.

On my side, it is interesting that I hit this ICE on our old Intel Broadwells with just AVX2. However, the build on our AMD EPYC 7282 (`znver2`)  worked fine and those also just support AVX2.